cmd/libs/go2idl/set-gen/main.go

```diff
@@ -0,0 +1,358 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// set-gen is an example usage of go2idl.
+package main
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"k8s.io/kubernetes/cmd/libs/go2idl/generator"
+	"k8s.io/kubernetes/cmd/libs/go2idl/namer"
+	"k8s.io/kubernetes/cmd/libs/go2idl/parser"
+	"k8s.io/kubernetes/cmd/libs/go2idl/types"
+
+	"github.com/golang/glog"
+	flag "github.com/spf13/pflag"
+)
+
+var (
+	inputDirs         = flag.StringP("input-dirs", "i", "./types", "comma-separated list of directories to get input types from.")
+	outputBase        = flag.StringP("output-base", "o", filepath.Join(os.Getenv("GOPATH"), "src"), "Output base; defaults to $GOPATH/src/")
```

Comment:
ARGH I did that right in one place and forgot about it here

On Thu, Oct 22, 2015 at 11:21 PM, Tim Hockin <notifications@github.com>
wrote:

> In cmd/libs/go2idl/set-gen/main.go
> <https://github.com/kubernetes/kubernetes/pull/15715#discussion_r42836212>
> :
>
> > +	"os"
> > +	"path/filepath"
> > +	"strings"
> > +
> > +	"k8s.io/kubernetes/cmd/libs/go2idl/generator"
> > +	"k8s.io/kubernetes/cmd/libs/go2idl/namer"
> > +	"k8s.io/kubernetes/cmd/libs/go2idl/parser"
> > +	"k8s.io/kubernetes/cmd/libs/go2idl/types"
> > +
> > +	"github.com/golang/glog"
> > +	flag "github.com/spf13/pflag"
> > +)
> > +
> > +var (
> > +	inputDirs         = flag.StringP("input-dirs", "i", "./types", "comma-separated list of directories to get input types from.")
> > +	outputBase        = flag.StringP("output-base", "o", filepath.Join(os.Getenv("GOPATH"), "src"), "Output base; defaults to $GOPATH/src/")
>
> not cool if GOPATH has multiple entries
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/15715/files#r42836212>.
>